transportable-udfs-annotation-processor/src/main/java/com/linkedin/transport/processor/TransportProcessor.java

```diff
@@ -0,0 +1,189 @@
+ */
+@SupportedOptions({"debug"})
+@SupportedAnnotationTypes({"*"})
+public class TransportProcessor extends AbstractProcessor {
+
+  private Types _types;
+  private Elements _elements;
+  private TypeMirror _topLevelStdUDFInterfaceType;
+  private TypeMirror _stdUDFClassType;
+  private UDFProperties _udfProperties;
+
+  @Override
+  public SourceVersion getSupportedSourceVersion() {
+    return SourceVersion.latestSupported();
+  }
+
+  @Override
+  public synchronized void init(ProcessingEnvironment processingEnv) {
+    super.init(processingEnv);
+    _types = processingEnv.getTypeUtils();
+    _elements = processingEnv.getElementUtils();
+    _topLevelStdUDFInterfaceType = _elements.getTypeElement(TopLevelStdUDF.class.getName()).asType();
+    _stdUDFClassType = _elements.getTypeElement(StdUDF.class.getName()).asType();
+    _udfProperties = new UDFProperties();
+  }
+
+  @Override
+  public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+    try {
+      processImpl(roundEnv);
+    } catch (Exception e) {
+      // We don't allow exceptions of any kind to propagate to the compiler
+      try (StringWriter stringWriter = new StringWriter(); PrintWriter printWriter = new PrintWriter(stringWriter)) {
+        e.printStackTrace(printWriter);
+        fatalError(stringWriter.toString());
+      } catch (IOException ioe) {
+        fatalError("Could not close resources " + ioe);
+      }
+    }
+    // Universal processors should return false since other processor can be potentially acting on the same element
+    return false;
+  }
+
+  private void processImpl(RoundEnvironment roundEnv) {
+    if (roundEnv.processingOver()) {
+      generateUDFPropertiesFile();
+    } else {
+      processElements(roundEnv.getRootElements());
+    }
+  }
+
+  private void processElements(Set<? extends Element> elements) {
+    for (Element element : elements) {
+      // Check if the element is a non-abstract class which extends the StdUDF class
+      if (element.getKind().equals(ElementKind.CLASS) && !element.getModifiers().contains(Modifier.ABSTRACT)
+          && _types.isAssignable(element.asType(), _stdUDFClassType)) {
+        processUDFClass((TypeElement) element);
+      }
+    }
+  }
+
+  /**
+   * Finds the {@link TopLevelStdUDF} for the given {@link StdUDF} class and adds it to the list of discovered UDFs
+   */
+  private void processUDFClass(TypeElement udfClassElement) {
+    debug(String.format("Processing UDF Class: %s", udfClassElement.getQualifiedName()));
+
+    TypeElement closestTopLevelStdUDFInterface = getClosestTopLevelStdUDFInterface(udfClassElement);
+    if (closestTopLevelStdUDFInterface != null) {
+      _udfProperties.addUDF(closestTopLevelStdUDFInterface.getQualifiedName().toString(),
+          udfClassElement.getQualifiedName().toString());
+    }
+  }
+
+  /**
+   * Returns a {@link TypeElement} for the closest (in the type hierarchy) class/interface which implements the
```

Comment:
closest class/interface (in the type hierarchy),

Reply:
Fixed